### ISSUE TITLE
RuleTester has errors after update to eslint 7.2

### ISSUE THREAD

**Issue by wenyuwang** [eslint/eslint#13414]
**Tell us about your environment**

* **ESLint Version:** 7.2.0
* **Node Version:** 12.17
* **npm Version:** 6.14.4

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports.rules = {
  'angular-component-controller-naming': angularComponentControllerNaming,
  'angular-component-lifecycle-functions': angularComponentLifecycleFunctions,
  'angular-invalid-nginject': angularInvalidNgInject,
  'angular-max-params': angularMaxParams,
  'jasmine-inject': jasmineInject,
  'jasmine-mocks': jasmineMocks,
  'toast-master-prefer-helper-functions': toastMasterPreferHelperFunctions,
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

1. we are updating eslint to 7.2.0 and other eslint dependencies to the latest version
2. And ruletester tests start failing 

```js
  function checkFunction(node) {
    const name = _.get(node, 'id.name') || _.get(node, 'parent.id.name') || node.type;

    if (!hasInjectAnnotation(node) && node.params.length > max) {
      context.report(
        node,
        `${name} has too many parameters (${node.params.length}). Maximum allowed is ${max}.`
      );
    }
  }

  function hasInjectAnnotation(node) {
    const comments = context.getComments(getAnnotationContainer(node));
    return _.some(
      comments.leading,
      (comment) => comment.type === 'Block' && _.trim(comment.value) === '@ngInject'
    );
  }

  function getAnnotationContainer(node) {
    if (node.parent.type === 'Property') {
      return node.parent;
    }
    return node.type === 'FunctionDeclaration' ? node : node.parent.parent;
  }

```

```bash
npx mocha  rules
```

**What did you expect to happen?**
1. Tests should not fail after updating.

**What actually happened? Please include the actual, raw output from ESLint.**
1. It  showed error messages like this 
```
     Error: Use token.range[0] instead of token.start
Occurred while linting <input>:3
      at Object.get (node_modules/eslint/lib/rule-tester/rule-tester.js:223:23)
      at SourceCode.getComments (node_modules/eslint/lib/source-code/source-code.js:353:50)
      at Object.getComments (node_modules/eslint/lib/linter/linter.js:810:96)
      at hasInjectAnnotation (rules/angular-max-params.js:21:30)
```

I suspected it might be related to the changes here - https://github.com/eslint/eslint/blame/master/lib/rule-tester/rule-tester.js#L223

because it seems to prefer using `range` instead of `start`, but source-code still using `start`
I am not sure if my understanding is correct or there is a workaround. 

- Maybe we can use the newer functions 🤔

**Are you willing to submit a pull request to fix this bug?**
Yes

**Comment by wenyuwang** [comment]
Resolved
